src/popup/components/TransactionList.vue

```diff
@@ -78,135 +89,166 @@ export default {
-                  || (isFungibleTokenTx(tr)
-                    && this.compareCaseInsensitive(tr.tx.type, SCHEMA.TX_TYPE.contractCall)
-                    && tr.recipient === address);
-              case 'tips':
-                return (tr.tx.contractId
-                  && (activeNetwork.tipContractV1 === tr.tx.contractId
-                    || activeNetwork.tipContractV2 === tr.tx.contractId)
-                  && (tr.tx.function === 'tip' || tr.tx.function === 'retip')) || tr.claim;
-              default:
-                throw new Error(`Unknown display mode type: ${this.displayMode.filter}`);
-            }
-          })
-          .filter(
-            (tr) => !this.searchTerm
-              || this.getTxSymbol(tr)
-                .toLocaleLowerCase()
-                .includes(this.searchTerm.toLocaleLowerCase()),
-          )
-          .sort((a, b) => {
-            const arr = [a, b].map((e) => new Date(e.microTime));
-            if (this.displayMode.rotated) arr.reverse();
-            return (arr[0] - arr[1]) || a.pending;
-          })
-          .slice(0, this.maxLength || Infinity);
-      },
-    }),
-    showSearchAndFilters() {
-      return (
-        this.showFilters
-        || this.displayMode.filter !== 'all'
-        || this.searchTerm
-      );
-    },
-  },
-  mounted() {
-    this.loadMore();
-    const polling = setInterval(() => this.getLatest(), 10000);
-    const appInner = document.querySelector('.app-inner');
-
-    appInner.addEventListener('scroll', this.checkLoadMore);
-    window.addEventListener('scroll', this.checkLoadMore);
-    this.$once('hook:destroyed', () => {
-      appInner.removeEventListener('scroll', this.checkLoadMore);
-      window.removeEventListener('scroll', this.checkLoadMore);
-      clearInterval(polling);
-      this.isDestroyed = true;
+  setup(props, { root }) {
+    const loading = ref(false);
+    const isDestroyed = ref(false);
+    const searchTerm = ref('');
+    const displayMode = ref({ rotated: true, filter: 'all', sort: 'date' });
+    const filters = ref({
+      all: {}, in: {}, out: {}, dex: {},
     });
-    this.$watch(({ displayMode }) => displayMode, this.checkLoadMore);
-  },
-  methods: {
-    checkLoadMore() {
-      if (this.isDestroyed || !this.transactions.nextPageUrl) return;
+
+    const availableTokens = computed(() => root.$store.state.fungibleTokens.availableTokens);
+    const transactions = useState<any>('transactions');
+
+    const getDexContracts = useGetter('getDexContracts');
+    const account = useGetter('account');
+    const activeNetwork = useGetter('activeNetwork');
+    const getAccountPendingTransactions = useGetter('getAccountPendingTransactions');
+    const getTxSymbol = useGetter('getTxSymbol');
+
+    function compareCaseInsensitive(
+      str1: string,
+      str2: string,
+    ) {
+      return str1.toLocaleLowerCase() === str2.toLocaleLowerCase();
+    }
```

Comment:
As it's not strictly related to TransactionList maybe you could move it to utils?

Reply:
i do this on my other pr